src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -566,12 +566,7 @@ export class TerminalService implements ITerminalService {
 	}
 
 	private async _onBeforeShutdownAsync(reason: ShutdownReason): Promise<boolean> {
-		if (this._backgroundedTerminalInstances.length) {
-			for (const terminal of this._backgroundedTerminalInstances.filter(b => !b.shouldPersist)) {
-				terminal.dispose();
-			}
-		}
-		if (this.instances.length === 0) {
+		if (this.instances.length === 0 && this._backgroundedTerminalInstances.length === 0) {
```

Comment:
Backgrounded terminals should never cause confirm on exit as that would confuse the user.